.github/workflows/ci.yml

```diff
@@ -43,10 +44,14 @@ jobs:
           # https://github.com/actions/runner/issues/409#issuecomment-752775072
           components: ${{ contains(matrix.channel, 'nightly') && 'miri' || '' }}
 
+    - name: Set feature string for cache key
+      run: |
+        echo "FEATURES_HASH=$(echo ${{ matrix.features }} | sha256sum | cut -d ' ' -f 1)" >> $GITHUB_ENV
```

Comment:
Can you add a comment here explaining why we need to do this?

Reply:
I wrote the following, let me know if you think I should mention something else

```md
    # The features string contains commas which cannot be part of the cache
    # key for the Rust Cache action. Instead, we hash the features
    # to get a string of legal characters.
```

Reply:
Perfect!